server/server.go

```diff
@@ -205,3 +234,9 @@ func setCorsHeaders(respw http.ResponseWriter) {
 	respw.Header().Set("Access-Control-Allow-Origin", "*")
 	respw.Header().Set("Access-Control-Allow-Headers", "Accept,Content-Type")
 }
+
+func (s *RpcEndPointServer) SetUnhealthy() {
+	s.healthyMu.Lock()
```

Comment:
**question (non-blocking):**

do we really need a whole mutex to protect something that is going to be changed only once?  (i.e. there will be only once that the transition from `true` to `false` will happen, no matter how many times `SetUnhealthy()` is called).

Reply:
I see what you mean, but I would better use mutex/atomic when you have separate threads every time. Because I don't know how compiler handles this. I vaguely remember that compilers can do something awful when handling situations like that. Especially bools.

Reply:
sure.  no way mutex could hurt here.

although, since `/health` is exposed to the internet, that might open a funny way to DDOS us via mutex contention.